cli/azd/pkg/azsdk/user_agent_policy_test.go

```diff
@@ -22,11 +24,11 @@ func TestOverrideUserAgent(t *testing.T) {
 		return mocks.CreateEmptyHttpResponse(request, http.StatusOK)
 	})
 
-	clientOptions := NewClientOptionsBuilderFactory(mockContext.HttpClient, expectedUserAgent, cloud.AzurePublic()).
-		NewClientOptionsBuilder().
-		BuildArmClientOptions()
-
-	client, err := armresources.NewClient("SUBSCRIPTION_ID", &mocks.MockCredentials{}, clientOptions)
+	client, err := armresources.NewClient("SUBSCRIPTION_ID", &mocks.MockCredentials{}, &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			PerCallPolicies: []policy.Policy{NewUserAgentPolicy(expectedUserAgent)},
```

Comment:
Added `Transport: mockContext.HttpClient` here, which I missed when coverting this.